### PR TITLE
Add hard_tanh activation

### PR DESCRIPTION
**Pull request by davidmascharka**
Also want to make sure this is the appropriate way to do the tests

### CODE REVIEW

File: src/mygrad/nnet/activations/hard_tanh.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+def hard_tanh(x, *, lower_bound=-1, upper_bound=1, constant=False):
+    """ Returns the hard hyperbolic tangent function.
+
+    The hard_tanh function is `lower_bound` where `x` <= `lower_bound`, `upper_bound` where
+    `x` >= `upper_bound`, and `x` where `lower_bound` < `x` < `upper_bound`.
+
+    Parameters
+    ----------
+    x : Union[numpy.ndarray, mygrad.Tensor]
```
Comment (davidmascharka) [inline src/mygrad/nnet/activations/hard_tanh.py]: should this be `array_like` instead?

Comment (rsokl) [reply src/mygrad/nnet/activations/hard_tanh.py]: Yep!
```suggestion
    x : array_like
```

File: src/mygrad/nnet/activations/hard_tanh.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        The upper bound on the hard tanh.
+
+    constant : boolean, optional (default=False)
+        If ``True``, the returned tensor is a constant (it
+        does not back-propagate a gradient).
+
+    Returns
+    -------
+    mygrad.Tensor
+        The result of applpying the hard tanh function elementwise to `x`.
```
Comment (rsokl) [inline src/mygrad/nnet/activations/hard_tanh.py]: I'm trying to make a habit of including an `Examples` section for each docstring - could you add one here?

File: src/mygrad/nnet/activations/hard_tanh.py (hunk 1 above)
Comment (rsokl) [inline src/mygrad/nnet/activations/hard_tanh.py]: Yep!

```suggestion
    x : array_like
```

**Pull request merged**